src/article.ts

```diff
@@ -182,46 +197,58 @@ const parseElement =
+        if (!elements) {
+            return [new Err('No body elements available')];
+        }
+        return elements.map(parseElement(docParser));
+    }
 
-const parseBlock = (docParser: DocParser) => (block: Block): LiveBlock =>
+const parseBlock = (docParser: DocParser) => (block: IBlock): LiveBlock =>
     ({
         id: block.id,
-        isKeyEvent: block.attributes.keyEvent,
-        title: block.title,
-        firstPublished: block.firstPublishedDate,
-        lastModified: block.lastModifiedDate,
+        isKeyEvent: block?.attributes?.keyEvent ?? false,
+        title: block?.title ?? "",
+        firstPublished: new Date(block?.firstPublishedDate?.iso8601 ?? 0),
+        lastModified: new Date(block?.lastModifiedDate?.iso8601 ?? 0),
         body: parseElements(docParser)(block.elements),
     })
 
-const parseBlocks = (docParser: DocParser) => (blocks: Block[]): LiveBlock[] =>
+const parseBlocks = (docParser: DocParser) => (blocks: IBlock[]): LiveBlock[] =>
     blocks.map(parseBlock(docParser));
 
 const articleFields = (docParser: DocParser, content: Content): ArticleFields =>
     ({
-        pillar: pillarFromString(content.pillarId),
-        headline: content.fields.headline,
-        standfirst: docParser(content.fields.standfirst),
-        byline: content.fields.byline,
-        bylineHtml: fromNullable(content.fields.bylineHtml).map(docParser),
-        publishDate: content.webPublicationDate,
+        pillar: pillarFromString(content?.pillarId),
+        headline: content?.fields?.headline ?? "",
+        standfirst: docParser(content?.fields?.standfirst ?? ""),
+        byline: content?.fields?.byline ?? "",
+        bylineHtml: fromNullable(content?.fields?.bylineHtml).map(docParser),
+        // eslint-disable-next-line
+        // @ts-ignore
+        // CAPI is sending us a string, even though the thrift definition is CapiDateTime
+        publishDate: content?.webPublicationDate,
         mainImage: articleMainImage(content).andThen(parseImage),
         contributors: articleContributors(content),
         series: articleSeries(content),
-        commentable: content.fields.commentable,
+        commentable: content?.fields?.commentable ?? false,
         tags: content.tags,
     })
 
-const articleFieldsWithBody = (docParser: DocParser, content: Content): ArticleFieldsWithBody =>
-    ({
+const articleFieldsWithBody = (docParser: DocParser, content: Content): ArticleFieldsWithBody => {
+    const body = content?.blocks?.body;
+    const elements = body?.length ? body[0]?.elements : undefined;
+    return ({
         ...articleFields(docParser, content),
-        body: parseElements(docParser)(content.blocks.body[0].elements),
+        body: parseElements(docParser)(elements),
     });
+}
 
 const fromCapi = (docParser: DocParser) => (content: Content): Article => {
-    switch (content.type) {
+    switch (content.type.toString()) {
```

Comment:
it feels like we should be able to use the type in the switch case? Is that not an enum?

Reply:
This is the problem where thrift is an enum with numbers, but the CAPI JSON returns strings :/

Reply:
So toString returns 0, 1, 2 etc?

Reply:
Actually it doesn't, but `content.type === ContentType.ARTICLE` doesn't work...

Reply:
I thought it was at first but not quite. It seems like this should be possible but it's not ``` content.type === ContentType.ARTICLE ```

Reply:
could you change the enum values to be strings? https://www.typescriptlang.org/docs/handbook/enums.html#string-enums

Reply:
These types were generated from this thrift file https://github.com/guardian/content-api-models/blob/master/models/src/main/thrift/content/v1.thrift#L25.

We didn't want to change any of the generated files but it might make sense to make these strings?